### PR TITLE
feat: set default requests and limits (#92)

### LINKED ISSUE #92: set default limits
**Describe the bug**
The default install does not set any limits or requests, which will make it fail when a ns has quotas.

```
  Warning  FailedCreate  17s (x7 over 2m58s)  replicaset-controller  (combined from similar events): Error creating: pods "pti-systemtest-redis-757955d999-xwbwb" is forbidden: failed quota: compute-resources: must specify limits.cpu,limits.memory,requests.cpu,requests.memory
```

**To Reproduce**
1. have quota for ns
2. create argo install
3. see that pods are not created

**Expected behavior**
Set default limits (which is seen as a good practice anyways)

**Screenshots**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**

> /kind enhancement

**What does this PR do / why we need it**:
#92 
GITOPS-823

**Have you updated the necessary documentation?**
No

* [x] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #92 
GITOPS-823

**How to test changes / Special notes to the reviewer**:
Create a namespace with resource quota as mentioned in the below link
https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/quota-memory-cpu-namespace/

Install argo cd from the openshift console in the namespace created above.

### CODE REVIEW

**Comment by wtam2018** [comment]
We still need to set the resource limit for:

- Redis HA Proxy
- ApplicationSet Controller
- Grafana

**Comment by iam-veeramalla** [comment]
> We still need to set the resource limit for:
> 
> * Redis HA Proxy
> * ApplicationSet Controller
> * Grafana

Blocked by https://github.com/argoproj-labs/argocd-operator/pull/281
@wtam2018 Can you please merge https://github.com/argoproj-labs/argocd-operator/pull/281 to unblock this PR

**Comment by varshab1210** [comment]
@iam-veeramalla  I am able to reproduce the ci unit test failure locally. I used the same test container which is used by openshift ci. Openshift ci each time uses a fresh test container https://github.com/redhat-developer/gitops-operator/blob/master/openshift-ci/build-root/Dockerfile . The steps I followed to reproduce it: 

```
$ sudo docker run -it registry.ci.openshift.org/openshift/release:golang-1.16

# cd ../..

# mkdir redhat-developer

# git clone https://github.com/redhat-developer/gitops-operator

#  cd gitops-operator

# git fetch origin pull/108/head:pull_108

# git checkout pull_108

# go mod vendor

# go mod tidy

# git status -s
```

cc_ @amitkrout

**Comment by iam-veeramalla** [comment]
> @iam-veeramalla I am able to reproduce the ci unit test failure locally. I used the same test container which is used by openshift ci. Openshift ci each time uses a fresh test container https://github.com/redhat-developer/gitops-operator/blob/master/openshift-ci/build-root/Dockerfile . The steps I followed to reproduce it:
> 
> ```
> $ sudo docker run -it registry.ci.openshift.org/openshift/release:golang-1.16
> 
> # cd ../..
> 
> # mkdir redhat-developer
> 
> # git clone https://github.com/redhat-developer/gitops-operator
> 
> # cd gitops-operator
> 
> # git fetch origin pull/108/head:pull_108
> 
> # git checkout pull_108
> 
> # go mod vendor
> 
> # go mod tidy
> 
> # git status -s
> ```
> 
> cc_ @amitkrout

@varshab1210 @amitkrout I think we need to correct this script. Test container should expect changes in go.sum file, if it executes `go mod tidy`

**Comment by varshab1210** [comment]
> > @iam-veeramalla I am able to reproduce the ci unit test failure locally. I used the same test container which is used by openshift ci. Openshift ci each time uses a fresh test container https://github.com/redhat-developer/gitops-operator/blob/master/openshift-ci/build-root/Dockerfile . The steps I followed to reproduce it:
> > ```
> > $ sudo docker run -it registry.ci.openshift.org/openshift/release:golang-1.16
> > 
> > # cd ../..
> > 
> > # mkdir redhat-developer
> > 
> > # git clone https://github.com/redhat-developer/gitops-operator
> > 
> > # cd gitops-operator
> > 
> > # git fetch origin pull/108/head:pull_108
> > 
> > # git checkout pull_108
> > 
> > # go mod vendor
> > 
> > # go mod tidy
> > 
> > # git status -s
> > ```
> > 
> > 
> > cc_ @amitkrout
> 
> @varshab1210 @amitkrout I think we need to correct this script. Test container should expect changes in go.sum file, if it executes `go mod tidy`

Yes, its reporting the changes done in go.sum file. Details - https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-developer_gito[…]-developer-gitops-operator-master-unit/1384452715693740032

**Review by amitkrout — CHANGES_REQUESTED** [review]
```
Create a namespace with resource quota as mentioned in the below link
https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/quota-memory-cpu-namespace/

Install argo cd from the openshift console in the namespace created above.
```

@iam-veeramalla Would you please script the above steps in e2e test file

**Comment by iam-veeramalla** [comment]
> ```
> Create a namespace with resource quota as mentioned in the below link
> https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/quota-memory-cpu-namespace/
> 
> Install argo cd from the openshift console in the namespace created above.
> ```
> 
> @iam-veeramalla Would you please script the above steps in e2e test file

Hi @amitkrout , There is a different story where this process will be automated. We can take the automation test creation as part of the same story
https://issues.redhat.com/browse/GITOPS-890

**Comment by amitkrout** [comment]
> Hi @amitkrout , There is a different story where this process will be automated. We can take the automation test creation as part of the same story
https://issues.redhat.com/browse/GITOPS-890

Sounds good!
I added the issue comment in the JIRA, so that we won't forget it :)

**Comment by iam-veeramalla** [comment]
@wtam2018  please review

**Review by amitkrout — DISMISSED** [review]

**Comment by iam-veeramalla** [comment]
@wtam Please have a look and suggest if any changed needed :)

we need this change to get merged to complete [GITOPS-890](https://issues.redhat.com/browse/GITOPS-890).

**Comment by iam-veeramalla** [comment]
Rebased with latest

File: pkg/controller/argocd/argocd.go
Comment (wtam2018) [inline pkg/controller/argocd/argocd.go:19]:
```suggestion
func getArgoApplicationSetSpec() *argoapp.ArgoCDApplicationSet {
```

File: pkg/controller/argocd/argocd.go
Comment (wtam2018) [inline pkg/controller/argocd/argocd.go:164]:
```suggestion
			ApplicationSet: getArgoApplicationSetSpec(),
```

**Pull request merged**
